storage/store.go

```diff
@@ -1199,6 +1199,15 @@ func (s *Store) Descriptor() (*proto.StoreDescriptor, error) {
 // method, args & reply into a Raft Cmd struct and executes the
 // command using the fetched range.
 func (s *Store) ExecuteCmd(ctx context.Context, call proto.Call) error {
+	return s.ExecuteCmdWithCallback(ctx, call, nil)
+}
+
+// ExecuteCmdWithCallback is called from ExecuteCmd. When a debug
+// callback is not nil, it will be called whenever the command is
+// executed (and retried).
+//
+// The method is exposed for testing.
+func (s *Store) ExecuteCmdWithCallback(ctx context.Context, call proto.Call, dbgCallback func()) error {
```

Comment:
We already have the `TestingCommandFilter`, for example:

		TestingCommandFilter = func(args proto.Request, reply proto.Response) bool {
			if _, ok := args.(*proto.ScanRequest); ok {
				count++
			}
			return false
		}

Could you use that instead of introducing this new debug hook?